File: XOutput.Client/Rest/Input/InputsClient.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace XOutput.Rest.Input
{
    public class InputsClient : HttpJsonClient
    {
        public InputsClient(IHttpClientProvider clientProvider) : base(clientProvider)
        {

        }

        public Task<IEnumerable<InputDeviceInfo>> GetInputsAsync(int timeoutMillis = 1000)
        {
            return GetAsync<IEnumerable<InputDeviceInfo>>("api/inputs", CreateToken(timeoutMillis));
        }

        public Task<InputDeviceInfo> GetInputAsync(string id, int timeoutMillis = 1000)
        {
            return GetAsync<InputDeviceInfo>("api/inputs/" + id, CreateToken(timeoutMillis));
        }
    }
}

File: Web/src/client/rest/input/InputsClient.ts
import { InputDeviceInfo } from '../../../api';
import { http } from '../http';

export const inputsClient = {
    getInputs(): Promise<InputDeviceInfo[]> {
        return http.get<InputDeviceInfo[]>("/inputs");
    },
    getInput(id: string): Promise<InputDeviceInfo> {
        return http.get<InputDeviceInfo>(`/inputs/${id}`);
    },
}
